File: Lab_9_MediatryPattern/project/src/CurrentWaqtWidget.java
public class CurrentWaqtWidget extends WidgetComponent
{
    private String currentWaqt;

    public CurrentWaqtWidget(Mediator mediator) {
        super(mediator);
    }

    public void updateCurrentWaqt(String waqt) {
        this.currentWaqt = waqt;
        System.out.println("Current Waqt Updated: " + currentWaqt);
    }

}

File: Lab9_Mosque_Management_System/Lab9_Mosque_Management_System/Widgets/CurrentWaqtWidget.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab9_Mosque_Management_System
{
    public class CurrentWaqtWidget : IWidget
    {
        private IMediator _mediator;

        public void SetMediator(IMediator mediator)
        {
            _mediator = mediator;
        }

        public void Update()
        {
            Console.WriteLine("Current Waqt updated based on Current Time and Waqt Time Tables.");
        }
    }
}
